### PR TITLE
Changes for rev1.0

### PR DESCRIPTION
**Pull request by asadate**
- Changed currentConsumption property into an array of objects since a component can have multiple supplies.
- Added into componentID.json a property to capture the digital datasheet specification revision referenced to generate an e-datasheet.
-  Initial assumption was that only supply pins required protections like OVP, UVLO. This is not the case. Expanded the definition of a powerSupply in ComponentProtectionThresholds.json file to include any pin (not just supply pins) that is monitored by protection mechanisms. Example, the CC lines in the USB Type-c PD controller may have over voltage protection.

### CODE REVIEW

File: part-spec/common/componentID.json [hunk 1]
```diff
@@ -106,11 +106,16 @@
                     "example": "2022-03-31",
                     "type": "string"
                 },
+                "eDataheetSpecRevision": {
+                    "description": "revision of the digital datasheet specifications used to create the digital datasheet",
+                    "example": "rev 1.0",
```
Comment (rsnancollas) [inline part-spec/common/componentID.json]: suggest that we make this an enum (where we add a string each time we release) - that way we won't get revisions that we can't match against anything.

so "enum": ["1.0"]

File: part-spec/ic_microcontroller/microcontroller.json [hunk 2]
```diff
@@ -53,9 +53,12 @@
                 "description": "firmware version of the part",
                 "type": "string"
             },
-            "currentConsumption": {
+             "currentConsumption": {
```
Comment (rsnancollas) [inline part-spec/ic_microcontroller/microcontroller.json]: why did you add a space?

**Review by rsnancollas — CHANGES_REQUESTED** [review]
mostly looks good - one change suggested for the eDatasheetSpecRevision.

File: part-spec/common/componentID.json (hunk 1 above)
Comment (asadate) [reply part-spec/common/componentID.json]: Changed

File: part-spec/ic_microcontroller/microcontroller.json (hunk 2 above)
Comment (asadate) [reply part-spec/ic_microcontroller/microcontroller.json]: changed

**Review by rsnancollas — APPROVED** [review]

File: part-spec/common/componentID.json [hunk 3]
```diff
@@ -106,11 +106,16 @@
                     "example": "2022-03-31",
                     "type": "string"
                 },
+                "eDataheetSpecRevision": {
+                    "description": "revision of the digital datasheet specifications used to create the digital datasheet",
```
Comment (ajayitem) [inline part-spec/common/componentID.json:110]: There's a spelling error here "eDataheetSpecRevision". It should be "eDatasheetSpecRevision"

Comment (asadate) [reply part-spec/common/componentID.json:110]: Good Catch. Changed

**Comment by asadate** [comment]
Changed currentConsumption property into an array of objects since a component can have multiple supplies.
Added into componentID.json a property to capture the digital datasheet specification revision referenced to generate an e-datasheet.
Initial assumption was that only supply pins required protections like OVP, UVLO. This is not the case. Expanded the definition of a powerSupply in ComponentProtectionThresholds.json file to include any pin (not just supply pins) that is monitored by protection mechanisms. Example, the CC lines in the USB Type-c PD controller may have over voltage protection.
